src/cryptonote_core/loki_name_system.cpp

```diff
@@ -227,222 +226,252 @@ static bool char_is_alphanum(char c)
   return result;
 }
 
-bool validate_lns_name_and_value(cryptonote::network_type nettype, uint16_t type, char const *name, int name_len, char const *value, int value_len, std::string *reason)
+bool validate_lns_name_and_value(cryptonote::network_type nettype, uint16_t type, char const *name, int name_len, char const *value, int value_len, lns_value *blob, std::string *reason)
 {
+  if (blob) *blob = {};
+  std::stringstream err_stream;
+  LOKI_DEFER { if (reason) *reason = std::move(err_stream.str()); };
```

Comment:
It's already an rvalue, no need to `std::move` it.

Reply:
Sirs. I believe the standard implements a copy `@return  A copy of one of the underlying sequences.` Kindly I agree to `std::move`.

Reply:
https://en.cppreference.com/w/cpp/io/basic_stringstream/str
`
The copy of the underlying string returned by str is a temporary object that will be destructed at the end of the expression, so directly calling c_str() on the result of str() (for example in auto *ptr = out.str().c_str();) results in a dangling pointer.
`

std::move unnecessary with std::string(std::string &&) constructor